### PR TITLE
add option to allow extraenv in esproxy service

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Description

Support allowing extraEnv values for es proxy service 

## Related Issues

https://github.com/astronomer/issues/issues/5559
https://github.com/astronomer/issues/issues/5544

## Testing

Do not merge this PR until this text is replaced with details about how these changes were tested.

## Merging

0.32.0

### CODE REVIEW

**Comment by pgvishnuram** [comment]
test cases pending

**Review by rbankston — APPROVED** [review]
Looks good to me

**Comment by pgvishnuram** [comment]
@rishkarajgi - Ford was facing some issue with their elasticsearch and they were not able to add proxy to this particular service - so we are now adding this option to allow custom env vars - static test case is pending

File: tests/chart_tests/test_external_elasticsearch.py [hunk 1]
```diff
@@ -462,3 +462,33 @@ def test_external_es_with_private_registry_enabled(self, kube_version):
             assert container["image"].startswith(
                 private_registry
             ), f"Container named '{name}' does not use registry '{private_registry}': {container}"
+
+    def test_externalelasticsearch_with_extraenv(self, kube_version):
+        """Test External ElasticSearch with secret passed from
+        config/values.yaml."""
```
Comment (danielhoherd) [inline tests/chart_tests/test_external_elasticsearch.py:468]:
```suggestion
        """Test External ElasticSearch with secret passed from config/values.yaml."""
```

File: tests/chart_tests/test_external_elasticsearch.py [hunk 2]
```diff
@@ -462,3 +462,33 @@ def test_external_es_with_private_registry_enabled(self, kube_version):
...
+                "charts/external-es-proxy/templates/external-es-proxy-deployment.yaml",
+            ],
+        )
+
+        assert len(docs) == 1
+        doc = docs[0]
+        assert doc["kind"] == "Deployment"
+        assert doc["apiVersion"] == "apps/v1"
+        assert doc["metadata"]["name"] == "release-name-external-es-proxy"
+        assert {"name": "NO_PROXY", "value": "127.0.0.1"} in doc["spec"]["template"][
```
Comment (danielhoherd) [inline tests/chart_tests/test_external_elasticsearch.py:492]:
```suggestion
        assert {"name": "TEST_VAR_NAME", "value": "test_var_value"} in doc["spec"]["template"][
```

**Review by danielhoherd — CHANGES_REQUESTED** [review]
Code LGTM. PR description needs to be completed.

**Pull request merged**